utils/src/check/clobber-functions.c

```diff
@@ -140,6 +140,57 @@ static struct clobber_function clobber_pb_qconf_wrong_version = {
 	&do_clobber_pb_qconf_wrong_version,
 };
 
+/*
+ * Pick a random non-empty quorum slot then returns a pointer to that slot
+ */
+static struct scoutfs_quorum_slot *pick_random_qslot(void)
+{
+	int j = 0;
+	int filled_slots[SCOUTFS_QUORUM_MAX_SLOTS];
+	int rand_slot;
+	uint16_t family;
+	srandom(time(NULL));
+
+	for (int i = 0; i < SCOUTFS_QUORUM_MAX_SLOTS; i++) {
+		family = le16_to_cpu(global_super->qconf.slots[i].addr.v4.family);
+		if (family == SCOUTFS_AF_NONE)
+			continue;
+
+		filled_slots[j] = i;
+		j++;
+	}
+	rand_slot = filled_slots[(random() % (j-1))];
+	return &global_super->qconf.slots[rand_slot];
```

Comment:
This doesn't actually work and causes a `Program received signal SIGFPE, Arithmetic exception.` because of the bug here.

First, you're correctly counting valid/used slots with `j` here,but then you use `% (j - 1)` but the modulo operator doesn't work like that. You need to `% j`. Then there's the bug that *none* of the quorum slots could actually be in use, and so your code could run and find `0` filled slots, and then do `rand_slot = filled_slots[random() % -1]` which I'm pretty sure is wildly out of bounds.

Basically, `X & Y` results in a value `between 0 and Y - 1`. If you have an array `[0,j - 1]`, then `random() % j` does the trick.

But you still have to validate that `j` is actually not `0` and within bounds.